sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+             sync_on_reconstruction is False
+          3) Create and return a new container otherwise
+
+        Returns:
+            str: name of the bucket(container)
+            bool: represents either or not the bucket is managed by skypilot
+
+        Raises:
+            StorageBucketCreateError: If creating the container fails
+            StorageBucketGetError: If fetching a container fails
+            StorageExternalDeletionError: If externally deleted container is
+                attempted to be fetched while reconstructing the Storage for
+                'sky storage delete' or 'sky start'
+        """
+        try:
+            container_url = data_utils.AZURE_CONTAINER_URL.format(
+                storage_account_name=self.storage_account_name,
+                container_name=self.name)
+            container_client = data_utils.create_az_client(
+                client_type='container', container_url=container_url)
+            if container_client.exists():
+                is_private = (True if container_client.get_container_properties(
+                )['public_access'] is None else False)
+                # when user attempts to use private container without
+                # access rights
+                if self.resource_group_name is None and is_private:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.StorageBucketGetError(
+                            _BUCKET_FAIL_TO_CONNECT_MESSAGE.format(
+                                name=self.name))
+                self._validate_existing_bucket()
+                return container_client.container_name, False
+            # when the container name does not exist under the provided
+            # storage account name and credentials, and user has the rights to
+            # access the storage account.
+            else:
+                if isinstance(self.source,
+                              str) and self.source.startswith('https://'):
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.StorageBucketGetError(
+                            'Attempted to use a non-existent bucket as a '
+                            f'source: {self.source}. Please check if the '
+                            'container name is correct.')
+        except azure.exceptions().ServiceRequestError as e:
+            # raised when storage account name to be used does not exist.
+            error_message = e.message
+            if 'Name or service not known' in error_message:
+                with ux_utils.print_exception_no_traceback():
+                    raise exceptions.StorageBucketGetError(
+                        'Attempted to fetch a bucket from non-existant '
+                        'storage account '
+                        f'name: {self.storage_account_name}. Please check '
+                        'if the name is correct.')
```

Comment:
same, how about the string not in the error message?

Reply:
Handled at [d58a10b](https://github.com/skypilot-org/skypilot/pull/3032/commits/d58a10b59c28049f2201cfce624149b4468530fa)